src/layers/conv.jl

```diff
@@ -118,29 +127,33 @@ function (l::GCNConv)(g::GNNGraph,
         end
     end
     Dout, Din = size(l.weight)
-    if Dout < Din
-        # multiply before convolution if it is more convenient, otherwise multiply after
-        x = l.weight * x
-    end
-    if edge_weight !== nothing
-        d = degree(g, T; dir = :in, edge_weight)
+
+    if g isa GNNHeteroGraph
+        # not the best practice but this loop is 
```

Comment:
@CarloLucibello I need your guidance how to change that part so that it can use correct edge type as an input instead of this dummy solution. The problem is that edge type is not passed to the GCNConv layer (only to HeteroConv) so I'm not sure how to add it.